tools/SchemaManager/Program.cs

```diff
@@ -63,7 +63,7 @@ private static ServiceProvider BuildServiceProvider(string[] args)
         // TODO: this won't work in OSS if the AuthenticationType is set to ManagedIdentity
         services.AddSingleton<ISqlConnectionBuilder, DefaultSqlConnectionBuilder>();
 
-        services.TryAddSingleton<SqlRetryLogicBaseProvider>(p =>
+        services.TryAddSingleton(p =>
```

Comment:
Don't we need to add the retryProvider before DefaultConnectionBuilder? We do that in `SqlServerBaseRegistrationExtensions`

Reply:
Order doesn't matter for registration of different services. It only makes for the same service type